### DISCUSSION TITLE
How to autostart after login?

### DISCUSSION THREAD

**Discussion by ayanamist** [ramensoftware/windhawk#5] [Q&A]
I'm using portable mode, i don't find a option to enable "autostart after login", is there any command-line switch to start minimized so i can create a shortcut by myself?

**Comment by m417z** [comment]
Yes, you can run `windhawk.exe` with the `-tray-only` command line switch.

**Reply by pjmdesi** [reply]
Thanks!
Is there any plan to make this a checkbox in the Windhawk portable settings someday?
Or does auto-adding a shortcut to the startup programs folder violate the spirit of a portable app?

**Reply by m417z** [reply]
No, as you said, the idea of a portable app is that no changes are made outside of its folder. You can violate that principle, but then why not use the standard (non-portable) version?

**Reply by ayanamist** [reply]
I use portable mode because i do not need to remember backup all settings before reinstall OS, which is required when using standard one. So i need this checkbox even in portable mode.

**Reply by m417z** [reply]
Well, that's another issue, having an export/import feature is planned.

**Reply by ayanamist** [reply]
No no no, export/import still require me to remember use it before reinstall OS, but i dont want to do this and usually will forget to do this.

**Reply by pjmdesi** [reply]
Yeah same thing for me. I like to go portable for every app possible (I have a folder full of them). When I reinstall Windows (which I find myself needing to do frequently because it's a POS), I can just use all the apps in that folder as I left them with no extra work. No installing, no remembering settings. The most I'd have to do is add to start menu (or startup folder like Windhawk). Maybe a button that says "add shortcut to startup folder" as part of first run?

**Reply by dpcdpc11** [reply]
> Yeah same thing for me. I like to go portable for every app possible (I have a folder full of them). When I reinstall Windows (which I find myself needing to do frequently because it's a POS), I can just use all the apps in that folder as I left them with no extra work. No installing, no remembering settings. The most I'd have to do is add to start menu (or startup folder like Windhawk). Maybe a button that says "add shortcut to startup folder" as part of first run?

+1 this! And plenty of portable apps give you the option to run at startup so I don't think users would mind adding an extra entry in the registry for this purpose, or you can go the shortcut route and @pjmdesi mentioned. Plus, there's no mention of the "-tray-only" parameter in the documentation, which is how I've ended in this thread.
Btw, thanks for making this app. Huge fan here. Been using your 7+ Taskbar Tweaker for years!

**Reply by Sc0tTyXL** [reply]
I use Windhawk in Portable mode on my cloud drive so that I can use the same settings on different machines. Would be great if there was an easy way to enable autostart. I think given enough information (changing something outside the folder) they can make an informed decision to enable it.

I finally updated to 24H2 and had to leave 7+TT after using that for many many years too. Migrated to Windhawk which has most of the features I need. Thanks for all the hard work all that time.

For any user who doesn't know how to manually do it:
1. Go to the Windhawk folder 
2. Create a shortcut of the windhawk.exe (right click menu) and give it a better name if you want
3. Open the properties of the shortcut (right click menu)
4. Change windhawk.exe to windhawk.exe -tray-only in the Target field.
5. Save changes
6. Open File Explorer
7. Go to %userprofile%\AppData\Roaming\Microsoft\Windows\Start Menu\Programs
8. Open the Startup folder (you can create it if it doesn't exist)
9. Move or copy the Shortcut into the Startup folder
10. Reboot

**Reply by eddoliver1** [reply]
Thanks! 💯

**Reply by on6hz** [reply]
> Yes, you can run `windhawk.exe` with the `-tray-only` command line switch.

I think I somehow disabled the tray icon from somewhere, but WOW this parameter work with no tray option, that does make sense but I love it.

**Reply by quietmansoath** [reply]
This didn't work for me and I'm not sure what part's wrong.

Got the following error:

<img width="586" height="544" alt="{8F8644DB-65EF-4A96-8EA2-826D35224AF2}" src="https://github.com/user-attachments/assets/0c5c0f02-8dc9-452d-8403-d04a1b3d6e73" />

**Reply by m417z** [reply]
@quietmansoath move `-tray-only` outside of the quotes.

**Comment by on6hz** [comment]
run: 
shell:startup

I sometimes backup this folder

**Reply by adi-works** [reply]
It works, but it's slow. Windows comes up with the default UI and it takes a bit of time until it loads up the portable executable.

**Reply by mmlovemk** [reply]
yes you are right it will take sometime
